src/goImport.ts

```diff
@@ -104,3 +107,52 @@ export function addImport(arg: string) {
 		}
 	});
 }
+
+export function addImportToWorkspace(arg: string) {
+	const editor = vscode.window.activeTextEditor;
+	const selection = editor.selection;
+
+	let importPath = '';
+	if (selection.isSingleLine) {
+		// Attempt to load a partial import path based on currently selected text
+		let selectedText = editor.document.getText(selection);
+		if (selectedText.length > 0) {
+			if (!selectedText.startsWith('"')) {
+				selectedText = '"' + selectedText;
+			}
+			if (!selectedText.endsWith('"')) {
+				selectedText = selectedText + '"';
+			}
+			importPath = getImportPath(selectedText);
+		}
+	}
+
+	if (importPath === '') {
+		// Failing that use the current line
+		let selectedText = editor.document.lineAt(selection.active.line).text;
+		importPath = getImportPath(selectedText);
+	}
+
+	if (importPath === '') {
+		vscode.window.showErrorMessage('No import path to add');
+		return;
+	}
+
+	let filePath = fixDriveCasingInWindows(editor.document.fileName);
+	let fileDirPath = path.dirname(filePath);
+	let currentWorkspace = getCurrentGoWorkspaceFromGOPATH(getCurrentGoPath(), fileDirPath);
```

Comment:
Take the case of multiple gopaths. Someone might have GOPATH set to `/Users/folderA;/Users/folderB`.
The current workspace is under /Users.folderA
The package in question could be under /Users/folderB
In this case, the current logic wont work.

How about passing the import path to `go list -f {{.Dir}}`? This would return the actual path for the package. You wouldnt need to to a `fs.exists` check either.

See `showPackageFiles` function in `goBrowsePackage.ts` for a reference on how to use `go list`.

Reply:
True, I'll sort this out.